applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_static_solver.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Check that applications were imported in the main script
+KratosMultiphysics.CheckRegisteredApplications("StructuralMechanicsApplication")
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+try:
+    # Check that applications were imported in the main script
+    KratosMultiphysics.CheckRegisteredApplications("MeshingApplication")
+    import KratosMultiphysics.MeshingApplication as MeshingApplication
+    missing_meshing_dependencies = False
+    missing_application = ''
+except ImportError as e:
+    missing_meshing_dependencies = True
+    # extract name of the missing application from the error message
+    import re
+    missing_application = re.search(r'''.*'KratosMultiphysics\.(.*)'.*''','{0}'.format(e)).group(1)
+
+# Import base class file
+import structural_mechanics_static_solver
+
+def CreateSolver(model, custom_settings):
+    return AdaptativeStaticMechanicalSolver(model, custom_settings)
+
+class AdaptativeStaticMechanicalSolver(structural_mechanics_static_solver.StaticMechanicalSolver):
```

Comment:
There is a lot of copy-pasting in this solver and the dynamic one (same for the solvers in contact) 
=> you could put some stuff into a utility or maybe use multiple inheritance (i.e. this solver would derive from `StaticMechanicalSolver` and e.g. from `BaseRemeshingSolver`)
This way you don't have to copy-paste a lot of things